tests/_client.lua

```diff
@@ -0,0 +1,33 @@
+local spawn = require("awful.spawn")
+
+-- This file provide a simple, yet flexible, test client.
+-- It is used to test the `awful.rules`
+
+return function(class, title)
+    local cmd = table.concat {
+        "echo \"",
+        "local lgi = require 'lgi';",
+        "local Gtk = lgi.require('Gtk');",
+        "local GdkPixbuf = lgi.GdkPixbuf;",
+        "Gtk.init();",
+        "local class = '",
+        class or 'test_app',"';",
+        "local window = Gtk.Window {",
+        "    default_width  = 100,",
+        "    default_height = 100,",
+        "    title          = '",
+            title or 'Awesome test client',
```

Comment:
Could you move this to the beginning of the function? `title = title or "awesome test client"` (and same for `class`)
(Because right now you handle `class = nil` in only one of two places)